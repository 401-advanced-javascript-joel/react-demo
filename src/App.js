import React from 'react';

import Header from './components/Header';
import Graphic from './components/Graphic';
import Content from './components/Content';
import picture from './pic.jpeg';
import picture1 from './pic1.jpeg';

import './styles/styles.scss';

function App() {
  let caption = 'Its a graphic';
  return (
    <div className='App'>
      <Header />
      <Content />
      <Graphic picture={picture} alt={caption} />
      <Graphic picture={picture1} alt={caption} />
    </div>
  );
}

export default App;
