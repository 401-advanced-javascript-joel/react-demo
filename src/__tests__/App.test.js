import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

describe('snapshot test for app', () => {
  test('matches the snapshot', () => {
    const currentHTML = renderer.create(<App />).toJSON();
    expect(currentHTML).toMatchSnapshot();
  });
});
