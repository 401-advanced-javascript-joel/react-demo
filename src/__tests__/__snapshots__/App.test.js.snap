// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for app matches the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="header"
  >
    <h1>
      Enter your name
    </h1>
    <input
      onChange={[Function]}
      type="text"
      value="Sarah Smalls"
    />
    <h2
      className="welcome-message"
    >
      Welcome 
      Sarah Smalls
      ! Do you like icecream?
    </h2>
    <p
      className="child"
    >
      Im a Child
    </p>
    <h2>
      This is my first React App
    </h2>
    <p>
      Coded by Joel Watson
    </p>
  </div>
  <div
    className="content"
  >
    <input
      onChange={[Function]}
      type="text"
    />
    <h3>
      Welcome 
      Sarah Smalls
    </h3>
  </div>
  <div
    className="graphic"
  >
    <img
      src="pic.jpeg"
    />
  </div>
  <div
    className="graphic"
  >
    <img
      src="pic1.jpeg"
    />
  </div>
</div>
`;
